src/python_testing/TC_IDM_4_2.py

```diff
@@ -198,9 +225,9 @@ async def test_TC_IDM_4_2(self):
         sub_cr1_step1.Shutdown()
 
         # *** Step 2 ***
-        self.print_step(2, "CR1 sends a subscription message to the DUT with MaxIntervalCeiling set to a value less than SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC. DUT sends a report data action to the CR1. CR1 sends a success status response to the DUT. DUT sends a Subscribe Response Message to the CR1 to activate the subscription.")
+        self.print_step(2, "CR1 sends a subscription message to the DUT with MaxIntervalCeiling set to a value less than subscription_max_interval_publisher_limit_sec. DUT sends a report data action to the CR1. CR1 sends a success status response to the DUT. DUT sends a Subscribe Response Message to the CR1 to activate the subscription.")
         min_interval_floor_sec = 1
-        max_interval_ceiling_sec = max(2, SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC - 5)
+        max_interval_ceiling_sec = max(2, subscription_max_interval_publisher_limit_sec - 5)
```

Comment:
Need some clarification:

An assert to prevent idleModeDuration to be 1 and fail the test if so?

If I do _max(0, ...)_ and the max turns out to be the 0 itself, that's what **max_interval_ceiling_sec** will be, but then **min_interval_floor_sec** is set to 1, which will make the **MIN** greater than the **MAX**, causing an error. Maybe I'm missing something.

Reply:
good point. Let's do the following then:
- add an assert that SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC >=1
- change this step to less than or equal to SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT_SEC. It's just trying to verify that the subscription maxinterval is correct.
- set the floor to 0.

Reply:
Done.